backend/context/context.py

```diff
@@ -82,17 +68,12 @@ def p2p_by_general(
 
     def p2p_by_travel_rule(
         self,
+        receiver_vasp_address: str,
         currency: str,
         amount: int,
-        receiver_vasp_address: str,
-        off_chain_reference_id: str,
+        metadata: bytes,
```

Comment:
How come this one we send in the fully formed metadata, and in `def p2p_by_general()` we form the metadata in the function? Wouldn't it be better to create the metadata as part of the function?

Reply:
Because offchain utils create the metadata for you in this case, hence simpler by passing in metadata. We can refactoring these methods later, current one is just to make minimum change for this PR which is big